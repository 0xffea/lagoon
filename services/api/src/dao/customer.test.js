const { Sql } = require('./customer');

describe('Sql', () => {
  describe('updateCustomer', () => {
    it('should create a proper query', () => {
      const cred = {
        permissions: {
          customers: [1, 2, 3]
        }
      };
      const input = {
        id: 1,
        patch: {
          name: "test"
        }
      }

      const ret = Sql.updateCustomer(cred, input);
      expect(ret).toMatchSnapshot();
    });
  });
  describe('selectCustomer', () => {
    it('should create a proper query', () => {
      const ret = Sql.selectCustomer(1);

      expect(ret).toMatchSnapshot();
    });
  });

});
