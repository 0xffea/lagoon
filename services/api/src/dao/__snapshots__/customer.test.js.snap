// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql selectCustomer should create a proper query 1`] = `"select * from \`customer\` where \`id\` = 1"`;

exports[`Sql updateCustomer should create a proper query 1`] = `"update \`customer\` set \`name\` = 'test' where \`id\` = 1 and \`id\` in (1, 2, 3)"`;
